ci/run-travis.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+#
+# Entry point for all travis builds, this will set up the Travis environment by
+# downloading any dependencies. It will then execute the `run.sh` script to
+# build and execute all tests.
+#
+# Much of this script was liberally stolen from rust-lang/libc
+#
+# Key variables that may be set from Travis:
+# - TRAVIS_RUST_VERSION: 1.1.0 ... stable/nightly/beta
+# - TRAVIS_OS_NAME: linux/osx
+# - DOCKER_IMAGE: posborne/rust-cross:arm
+# - TARGET: e.g. arm-unknown-linux-gnueabihf
+
+set -ex
+
+BASE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+
+if [ "$TRAVIS_OS_NAME" = "linux" ]; then
+  OS=unknown-linux-gnu
+else
+  OS=apple-darwin
```

Comment:
Sure.  This bit was taken from libc, but I agree that it is probably better to be a bit more strict: https://github.com/rust-lang/libc/blob/master/ci/run-travis.sh#L11